mesatee_services/tdfs/internal/client/src/tdfs_client.rs

```
@@ -143,8 +144,13 @@ impl TDFSClient {
 
         let target = config::Internal::target_kms();
         let mut client = KMSClient::new(target)?;
-        let key_resp = client.request_get_key(&file_info.key_id)?;
-        let key_config = key_resp.config;
+        let key_req = kms_proto::proto::GetKeyRequest::new(&file_info.key_id);
+        let key_resp = client.get_key(key_req)?;
+        let key_config = key_resp.get_key_config()?;
+        let key_config = match key_config {
+            kms_proto::KeyConfig::Aead(config) => kms_proto::proto::AeadConfig::from(config),
+            kms_proto::KeyConfig::ProtectedFs(_config) => unimplemented!(), // ProtectedFS is not used by TDFS yet. Config of ProtectedFs will not be generated neither.
```

shall we return an error here?